python/instrumentation/openinference-instrumentation-litellm/tests/test_instrumentor.py

```diff
@@ -0,0 +1,319 @@
+        assert span.attributes["input.messages.1.role"] == "assistant"
+        assert span.attributes["input.messages.2.role"] == "user"
+
+    async def test_acompletion(self, tracer_provider, in_memory_span_exporter, instrumentor):
+        in_memory_span_exporter.clear()
+
+        await litellm.acompletion(
+            model="gpt-3.5-turbo",
+            messages=[{"content": "What's the capital of China?", "role": "user"}],
+            mock_response="Beijing",
+        )
+        spans = in_memory_span_exporter.get_finished_spans()
+        assert len(spans) == 1
+        span = spans[0]
+
+        assert span.name == "acompletion"
+        self._assert_common_span_attributes_for_completion(
+            span, "gpt-3.5-turbo", "What's the capital of China?", "Beijing"
+        )
+
+    def test_completion_with_retries(self, tracer_provider, in_memory_span_exporter, instrumentor):
+        in_memory_span_exporter.clear()
+
+        litellm.completion_with_retries(
+            model="gpt-3.5-turbo",
+            messages=[{"content": "What's the capital of China?", "role": "user"}],
+            mock_response="Beijing",
+        )
+        spans = in_memory_span_exporter.get_finished_spans()
+        assert len(spans) == 1
+        span = spans[0]
+
+        assert span.name == "completion_with_retries"
+        self._assert_common_span_attributes_for_completion(
+            span, "gpt-3.5-turbo", "What's the capital of China?", "Beijing"
+        )
+
+    # Bug report filed on GitHub for acompletion_with_retries: https://github.com/BerriAI/litellm/issues/4908
+    # Until litellm fixes acompletion_with_retries keep this test commented
+    # async def test_acompletion_with_retries(self, tracer_provider, in_memory_span_exporter):
+    #     in_memory_span_exporter.clear()
+
+    #     await litellm.acompletion_with_retries(
+    #         model="gpt-3.5-turbo",
+    #         messages=[{"content": "What's the capital of China?", "role": "user"}],
+    #         mock_response="Beijing",
+    #     )
+    #     spans = in_memory_span_exporter.get_finished_spans()
+    #     assert len(spans) == 1
+    #     span = spans[0]
+    #     assert span.name == "acompletion_with_retries"
+    #     self._assert_common_span_attributes_for_completion(span,
+    #                                                        "gpt-3.5-turbo",
+    #                                                        "What's the capital of China?",
+    #                                                        "Beijing")
+
+    # Unlike the completion() functions, liteLLM does not offer a mock_response parameter
+    # for embeddings or image gen yet
+    # For now the following tests monkeypatch OpenAIChatCompletion functions
+
+    def test_embedding(self, tracer_provider, in_memory_span_exporter, instrumentor):
+        in_memory_span_exporter.clear()
+
+        with patch.object(
+            OpenAIChatCompletion, "embedding", return_value=self.mock_response_embedding
```

Comment:
similarly here, we can just define `mock_response_embedding` directly in this function so we can see everything just in this one function, tests are important debugging tools and having everything in one place is quite useful IMO

Reply:
done, moved mock_response_embedding and mock_response_image_gen into corresponsing test funcs